src/browser/HostInstaller.h

```diff
@@ -0,0 +1,63 @@
+/*
+*  Copyright (C) 2017 Sami Vänttinen <sami.vanttinen@protonmail.com>
+*  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+*
+*  This program is free software: you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation, either version 3 of the License, or
+*  (at your option) any later version.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef HOSTINSTALLER_H
+#define HOSTINSTALLER_H
+
+#include <QObject>
+#include <QJsonObject>
+#include <QSettings>
+
+class HostInstaller : public QObject
+{
+    Q_OBJECT
+
+public:
+    enum SupportedBrowsers : int {
+        CHROME =    0,
+        CHROMIUM =  1,
+        FIREFOX =   2,
+        VIVALDI =   3
+    };
+
+public:
+    HostInstaller();
+    bool checkIfInstalled(SupportedBrowsers browser);
+    void installBrowser(SupportedBrowsers browser, const bool enabled, const bool proxy = false, const QString location = QString());
```

Comment:
Must have missed that before: Use const& instead of const value for non-integral parameters. Same for other methods in this class. Also don't default-initialize with `QString()`. Do that with `""`.